doc/faq.markdown

```diff
@@ -1,11 +1,19 @@
 Frequently Asked Questions {#faq}
 ==========================
 
--   Q: Example question?
+-   **What is InputArray?**
 
-    A: Example answer
+    It can be seen that almost all OpenCV functions receive InputArray type. What is it, and how can I understand the actual input types of parameters?
 
 
--   Q: Example question?
+    This is the proxy class for passing read-only input arrays into OpenCV functions.
 
-    A: Example answer
+    Inside a function you should use _InputArray::getMat() method to construct a matrix header for the array (without copying data). _InputArray::kind() can be used to distinguish Mat from vector<> etc., but normally it is not needed.
```

Comment:
It may be better to use `InputArray` instead of `_InputArray`, as most functions use `InputArray`.